orchagent/fabricportsorch.cpp

```diff
@@ -836,6 +837,185 @@ void FabricPortsOrch::updateFabricDebugCounters()
     }
 }
 
+void FabricPortsOrch::updateFabricCapacity()
+{
+    // Init value for fabric capacity monitoring
+    int capacity = 0;
+    int downCapacity = 0;
+    string lnkStatus = "down";
+    string configIsolated = "0";
+    string isolated = "0";
+    string autoIsolated = "0";
+    int operating_links = 0;
+    int total_links = 0;
+    int threshold = 100;
+    std::vector<FieldValueTuple> constValues;
+    string applKey = FABRIC_MONITOR_DATA;
+
+    // Get capacity warning threshold from APPL_DB table FABRIC_MONITOR_DATA
+    // By default, this threshold is 100 (percentage).
+    bool cfgVal = m_applMonitorConstTable->get("FABRIC_MONITOR_DATA", constValues);
+    if(!cfgVal)
+    {
+        SWSS_LOG_INFO("%s default values not set", applKey.c_str());
+    }
+    else
+    {
+        SWSS_LOG_INFO("%s has default values", applKey.c_str());
+    }
+    string configVal = "1";
+    for (auto cv : constValues)
+    {
+        configVal = fvValue(cv);
+        if (fvField(cv) == "monCapacityThreshWarn")
+        {
+            threshold = stoi(configVal);
+            SWSS_LOG_INFO("monCapacityThreshWarn: %s %s", configVal.c_str(), fvField(cv).c_str());
+            continue;
+        }
+    }
+
+    // Check fabric capacity.
+    SWSS_LOG_INFO("FabricPortsOrch::updateFabricCapacity start");
+    for (auto p : m_fabricLanePortMap)
+    {
+        int lane = p.first;
+        string key = FABRIC_PORT_PREFIX + to_string(lane);
+        std::vector<FieldValueTuple> values;
+        string valuePt;
+
+        // Get fabric serdes link status from STATE_DB
+        bool exist = m_stateTable->get(key, values);
+        if (!exist)
+        {
+            SWSS_LOG_INFO("No state infor for port %s", key.c_str());
+            return;
+        }
+        for (auto val : values)
+        {
+            valuePt = fvValue(val);
+            if (fvField(val) == "STATUS")
+            {
+                lnkStatus = valuePt;
+                continue;
+            }
+            if (fvField(val) == "CONFIG_ISOLATED")
+            {
+                configIsolated = valuePt;
+                continue;
+            }
+            if (fvField(val) == "ISOLATED")
+            {
+                isolated = valuePt;
+                continue;
+            }
+            if (fvField(val) == "AUTO_ISOLATED")
+            {
+                autoIsolated = valuePt;
+                continue;
+            }
+        }
+       // Calculate total number of serdes link, number of operational links,
+       // total fabric capacity.
+        bool linkIssue = false;
+        if (configIsolated == "1" || isolated == "1" || autoIsolated == "1")
+        {
+            linkIssue = true;
+        }
+
+        if (lnkStatus == "down" || linkIssue == true)
+        {
+            downCapacity += 44316;
+        }
+        else
+        {
+            capacity += 44316;
```

Comment:
where is  is `44316` coming from?

Reply:
that's the  90% of the enocdded serdes rate .